examples/karpenter/main.tf

```diff
@@ -104,16 +145,17 @@ module "eks_blueprints" {
 module "eks_blueprints_kubernetes_addons" {
   source = "../../modules/kubernetes-addons"
 
-  eks_cluster_id           = module.eks_blueprints.eks_cluster_id
-  eks_cluster_endpoint     = module.eks_blueprints.eks_cluster_endpoint
-  eks_oidc_provider        = module.eks_blueprints.oidc_provider
-  eks_cluster_version      = module.eks_blueprints.eks_cluster_version
-  auto_scaling_group_names = module.eks_blueprints.self_managed_node_group_autoscaling_groups
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider    = module.eks_blueprints.oidc_provider
+  eks_cluster_version  = module.eks_blueprints.eks_cluster_version
 
   enable_karpenter                    = true
   enable_aws_node_termination_handler = true
 
   tags = local.tags
+
+  depends_on = [module.eks_blueprints.managed_node_group_autoscaling_groups]
```

Comment:
we want to avoid hard dependencies because any update to the managed node group will force the computed attributes in addons to be re-computed. This forces IRSA policies to be detached and re-attached but only after some time which results in failure to provision correctly

Reply:
Ok will remove it